junit-jupiter-api/src/main/java/org/junit/jupiter/api/AutoCloseExtension.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2015-2023 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.api;
```

Comment:
FYI: having this in the `api` package unfortunately introduces a package cycle between `api` and `api.extension`. 😞 

So I've reverted to the "built-in extension" approach in my branch.